test/App/Fossa/Report/AttributionSpec.hs

```diff
@@ -0,0 +1,77 @@
+module App.Fossa.Report.AttributionSpec
+  ( spec,
+  )
+where
+
+import App.Fossa.Report.Attribution
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Prologue
+import Test.Hspec
+import Test.Hspec.Hedgehog
+
+defaultRange :: Range Int
+defaultRange = Range.linear 0 50
+
+genLicense :: Gen License
+genLicense =
+  License
+    <$> (LicenseName <$> arbitraryText)
+    <*> (LicenseContents <$> arbitraryText)
+
+genProject :: Gen Project
+genProject = Project <$> arbitraryText <*> arbitraryText
+
+genDependency :: Gen Dependency
+genDependency =
+  Dependency
+    <$> arbitraryText
+    <*> arbitraryText
+    <*> arbitraryText
+    <*> Gen.bool
+    <*> arbitraryText
+    <*> Gen.list defaultRange arbitraryText
+    <*> Gen.maybe arbitraryText
+    <*> Gen.list defaultRange genLicense
+    <*> Gen.list defaultRange genLicense
+    <*> arbitraryText
+    <*> Gen.list defaultRange arbitraryText
+    <*> Gen.list defaultRange arbitraryText
+    <*> arbitraryText
+    <*> arbitraryText
+
+genAttribution :: Gen Attribution
+genAttribution =
+  Attribution
+    <$> genProject
+    <*> Gen.list defaultRange genDependency
+    <*> Gen.list defaultRange genDependency
+    <*> genLicenseMap
+
+tuplify :: Monad m => m a -> m b -> m (a, b)
+tuplify a b = do
```

Comment:
[nit] this is the same as `(,) <$> a <*> b` with applicative syntax
or `liftA2 (,) a b`